bertopic/backend/_spacy.py

```diff
@@ -76,19 +82,20 @@ def embed(self,
         # Handle empty documents, spaCy models automatically map
         # empty strings to the zero vector
         empty_document = " "
+        embeddings = []
 
-        # Extract embeddings from a transformer model
-        if "transformer" in self.embedding_model.component_names:
-            embeddings = []
-            for doc in tqdm(documents, position=0, leave=True, disable=not verbose):
-                embeddings.append(self.embedding_model(doc or empty_document)._.trf_data.tensors[-1][0].tolist())
-            embeddings = np.array(embeddings)
+        for doc in tqdm(documents, position=0, leave=True, disable=not verbose):
+            spacy_embedding = self.emebdding_model(doc or empty_document)
 
-        # Extract embeddings from a general spacy model
-        else:
-            embeddings = []
-            for doc in tqdm(documents, position=0, leave=True, disable=not verbose):
-                embeddings.append(self.embedding_model(doc or empty_document).vector)
-            embeddings = np.array(embeddings)
+            # Extract embeddings from a transformer model
+            if "transformer" in self.embedding_model.component_names:
+                embedding = spacy_embedding._.trf_data.tensors[-1][0]
+                embedding = self._get_if_cupy(embedding).tolist()
+
+            # Extract embeddings from a general spacy model
+            else:
+                embedding = self._get_if_cupy(embedding.vector)
```

Comment:
We could simplify this more if we were wiling to also call `tolist()` on the non-transformer approach. I don't know the background on why only the transformer one had that called on it previously though :thinking: